src/leo_s3_bucket.erl

```diff
@@ -903,35 +930,33 @@ find_bucket_by_name_2(Bucket, DB, Node, Value) ->
     RPCKey = rpc:async_call(Node, leo_s3_bucket, find_bucket_by_name,
                             [Bucket, LastModifiedAt]),
     Ret = case rpc:nb_yield(RPCKey, ?DEF_REQ_TIMEOUT) of
-              {value, {ok, match}} when Value == null ->
-                  not_found;
-              {value, {ok, match}} when Value /= null ->
-                  {ok, Value};
-              {value, {ok, Value_1}} ->
-                  NewBucketVal = Value_1#?BUCKET{last_synchroized_at = leo_date:now()},
-                  case DB of
-                      mnesia ->
-                          catch leo_s3_bucket_data_handler:insert(
-                                  {DB, ?BUCKET_TABLE},
-                                  Value#?BUCKET{del = true,
-                                                last_modified_at = leo_date:now()
-                                               });
-                      _ ->
-                          catch leo_s3_bucket_data_handler:delete(
-                                  {DB, ?BUCKET_TABLE}, Value)
-                  end,
-                  leo_s3_bucket_data_handler:insert({DB, ?BUCKET_TABLE}, NewBucketVal),
-                  {ok, NewBucketVal};
-              {value, not_found} ->
-                  not_found;
-              _ when Value == null ->
-                  not_found;
-              _ ->
-                  {ok, Value}
-          end,
+        {value, {ok, match}} when Value == null ->
+            not_found;
+        {value, {ok, match}} when Value /= null ->
+            {ok, Value};
+        {value, {ok, Value_1}} ->
+            {ok, Value_1};
+        {value, not_found} ->
+            not_found;
+        _ when Value == null ->
+            not_found;
+        timeout ->
+            {error, timeout};
+        {badrpc, Reason} ->
+            {error, Reason}
+    end,
+    case Ret of
+        {error, Cause} ->
+            error_logger:warning_msg("~p,~p,~p, Synchronization of Bucket Data Failed, Node:~p, Reason:~p~n",
+                                     [{module, ?MODULE_STRING}, 
+                                      {function, "find_bucket_by_name_2/3"},
+                                      {line, ?LINE},
+                                      Node, Cause]);
```

Comment:
@windkit 
Must bind Ret with {error, Cause}.
if not, retrying against leo_manager slave will not occur.